source/features/remove-projects.js

```diff
@@ -0,0 +1,42 @@
+import select from 'select-dom';
+import * as pageDetect from '../libs/page-detect';
+
+const projectsTabSelector = '.js-repo-nav .reponav-item[data-selected-links^="repo_projects"]';
+
+function removeProjectsTab() {
+	const projectsTab = select(projectsTabSelector);
+
+	if (projectsTab && projectsTab.querySelector('.Counter, .counter').textContent === '0') {
+		projectsTab.remove();
+	}
+}
+
+function removeProjectsFilter() {
+	const filters = select.all('.issues-listing .table-list-filters .select-menu-button');
+	const [projectsFilter] = filters.filter(filter => filter.textContent.includes('Projects'));
+
+	projectsFilter.remove();
+}
+
+function removeProjectsSidebarItem() {
+	const sidebarItems = select.all('.js-discussion-sidebar-item');
+	const [projectsSidebarItem] = sidebarItems.filter(item => select('.discussion-sidebar-heading', item).textContent.includes('Projects'));
+
+	projectsSidebarItem.remove();
+}
+
+export default function () {
+	removeProjectsTab();
+
+	if (select.exists(projectsTabSelector)) {
```

Comment:
In #977, I mentioned that the projects tab in the repo is not the authoritative source whether you have projects in the selector or not.

> There are both repository and organisation projects in there so we should check the contents of the selector instead of using the number of projects in the repo as we do for the repository tab.

If you have zero projects in the repository, but you have organisation projects, the  repository tab counter in the would still say `0` and we would remove the tab. But the projects selector would still hold projects in the organisation tab which could be used.

![image](https://user-images.githubusercontent.com/506129/34868096-9acad10e-f77a-11e7-8221-a9eebb4ebb69.png)

Reply:
Good catch… I will change the checks. :+1:

Reply:
You can't just _change_ them, in most cases there's no way to tell if:

- the issue is in an org
- there are projects in the org

I looked in the HTML of a few pages and only in the _new issue_ page I found my organization project.

In short, sadly I think that in most cases it can't be hidden

Reply:
@bfred-it Hmm, before I've created the issue, I've checked and didn't see an XHR request for the projects, only for the labels. Could it have been cached on the client?

Reply:
It happens on hover

![demo](https://user-images.githubusercontent.com/1402241/34874050-06f0e1b2-f7ca-11e7-975f-1063b6fafdf4.gif)